### ISSUE TITLE
Support additional SatlasPretrain models.

### ISSUE THREAD

**Issue by piperwolters** [microsoft/torchgeo#1861]
### Summary

I see that a couple SatlasPretrain models [have been merged](https://github.com/microsoft/torchgeo/pull/1358/files) and will be included in the 0.6.0 release, which is great!
I'm hoping that additional pretrained weights and models could be integrated, specifically the Sentinel-1 and Landsat models as well as models with varying number of input images and bands. 

We recently released the [satlaspretrain-models package](https://pypi.org/project/satlaspretrain-models/) which makes it easy to load the various pretrained models. I could make a PR to torchgeo to include more pretrained models, either 1) using the code that was put in place in the above PR or 2) using this new satlaspretrain-models package. Wondering if you all have a preference?
Thanks!

**Comment by adamjstewart** [comment]
Yes, we would love to get the rest of the SatlasPretrain models in TorchGeo so they're as easy as possible for our users to access!

My preference would be for option 1. This avoids adding "yet another dependency" and integrates nicely with our Lightning trainers. Note that we've really only added backbones in TorchGeo so far. We would like to add decoders as well, but haven't yet put a lot of thought into how the API should work.

P.S. When we added the first couple SatlasPretrain models, we assumed they were under the Apache-2.0 license, but based on this new repo it seems they are actually ODC-BY. Is this correct? I'll update our documentation if so.

FYI @favyen2

**Comment by piperwolters** [comment]
Okay, sounds good. I will just start by adding the additional backbones in a PR, and if you all come up with a plan or skeleton for decoders, I can make a PR for that later.

And yes, our data and model checkpoints are under ODC-BY and the code is Apache 2.0.
Thank you!

**Comment by isaaccorley** [comment]
@piperwolters I added the previous Satlas checkpoints. If you need any help let me know! We are also active on the TorchGeo slack channel if you want to discuss.

**Comment by piperwolters** [comment]
@adamjstewart That first PR only added single-image Swin backbones. Would like to add multi-image backbones and decoders eventually, but don't necessarily need to keep this issue open for that.

**Comment by adamjstewart** [comment]
Nah, let's keep it open. You could add a bullet point list to the issue like:

- [x] Swin Sentinel single-image
- [ ] Swin Sentinel multi-image

if you want to track progress on completion.

**Comment by adamjstewart** [comment]
Additional remaining backbones added in #2038

**Comment by adamjstewart** [comment]
We added OlmoEarth decoders recently, do we want to add Satlas decoders as well?
